classes/local/object_manipulator/candidates/candidates_factory.php

```diff
@@ -0,0 +1,72 @@
+<?php
+// This file is part of Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+/**
+ * Class candidates_factory
+ * @package tool_objectfs
+ * @author Gleimer Mora <gleimermora@catalyst-au.net>
+ * @copyright Catalyst IT
+ * @license http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+
+namespace tool_objectfs\local\object_manipulator\candidates;
+
+use moodle_exception;
+use stdClass;
+use tool_objectfs\local\object_manipulator\checker;
+use tool_objectfs\local\object_manipulator\deleter;
+use tool_objectfs\local\object_manipulator\puller;
+use tool_objectfs\local\object_manipulator\pusher;
+use tool_objectfs\local\object_manipulator\recoverer;
+
+defined('MOODLE_INTERNAL') || die();
+
+class candidates_factory {
+
+    /** @var manipulator_candidates_base $finder  */
+    private $finder;
+
+    /** @var array $manipulatormap */
+    private $manipulatormap = [
+        checker::class => checker_candidates::class,
+        deleter::class => deleter_candidates::class,
+        puller::class => puller_candidates::class,
+        pusher::class => pusher_candidates::class,
+        recoverer::class => recoverer_candidates::class,
+    ];
+
+    /**
+     * candidates_factory constructor.
+     * @param string $manipulator
+     * @param stdClass $config
+     * @throws moodle_exception
+     */
+    public function __construct($manipulator, stdClass $config) {
```

Comment:
I don't see what we gain by instantiating this class, i would think about just making the finder method static + removing the constructor

Reply:
You are right, in this case it seems to be a good idea, but personally I'm not a friend of static methods, I try to avoid them if possible. I've update it to use `static method`.